src/mono/wasm/build/WasmApp.targets

```diff
@@ -122,6 +124,10 @@
       <WasmMainAssemblyFileName Condition="'$(WasmMainAssemblyFileName)' == ''">$(TargetFileName)</WasmMainAssemblyFileName>
 
       <WasmAppDir>$([MSBuild]::NormalizeDirectory($(WasmAppDir)))</WasmAppDir>
+
+      <_MainAssemblyPath Condition="'%(WasmAssembliesToBundle.FileName)' == $(AssemblyName) and '%(WasmAssembliesToBundle.Extension)' == '.dll'">%(WasmAssembliesToBundle.Identity)</_MainAssemblyPath>
```

Comment:
We should do this if `$(WasmGenerateAppBundle) == 'true'`. If the assembly is not found, then we should emit a warning.

Reply:
Ah, that's the condition that I was missing...